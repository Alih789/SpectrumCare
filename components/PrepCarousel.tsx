import React from 'react';
import {
  View,
  Text,
  Dimensions,
  ScrollView,
  StyleSheet,
  Image,
} from 'react-native';
import Carousel from 'react-native-reanimated-carousel';
import 'react-native-reanimated';
import {Header} from 'react-native/Libraries/NewAppScreen';

type PrepCarouselProps = {
  imageURLs: string[];
  headers: string[];
  bodyText: string[];
};

const {width, height} = Dimensions.get('window');

export default function PrepCarousel({
  imageURLs,
  headers,
  bodyText,
}: PrepCarouselProps): JSX.Element {
  let JSXData = [];
  for (const i in imageURLs) {
    JSXData.push(
      <>
        <Text style={styles.header}>{headers[i]}</Text>
        <Image
          key={imageURLs[i]}
          source={{uri: imageURLs[i]}}
          style={styles.image}
        />
        <Text key={bodyText[i]} style={styles.text}>
          {bodyText[i]}
        </Text>
      </>,
    );
  }

  return (
    <ScrollView
      style={styles.scrollView}
      contentContainerStyle={styles.contentContainer}>
      <Carousel
        loop={false}
        width={width}
        height={height}
        data={JSXData}
        panGestureHandlerProps={{
          activeOffsetX: [-10, 10],
          activeOffsetY: [-10, 10],
        }}
        scrollAnimationDuration={1000}
        onSnapToItem={(index: number) => console.log('current index:', index)}
        renderItem={({item}) => <View>{item}</View>}
      />
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  contentContainer: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  header: {
    color: 'white',
    fontSize: 25,
  },
  image: {
    resizeMode: 'cover',
    height: height / 2,
    width: width,
  },
  scrollView: {
    alignSelf: 'center',
    borderColor: 'black',
    margin: 15,
  },

  text: {
    color: 'white',
    fontSize: 15,
  },
});
