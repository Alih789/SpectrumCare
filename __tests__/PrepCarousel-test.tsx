import 'react-native';
import React from 'react';
import PrepCarousel from '../components/PrepCarousel';
import renderer from 'react-test-renderer';


describe("PrepCarousel", () => {
    const procedureInfo = require('../assets/testData/ProcedureData.json');

    test("renders PrepCarousel", () => {
        const tree = renderer.create(
            <PrepCarousel
            procedureInfo={procedureInfo['mri']}
            data-testid="container-test-id"
            />
            ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});