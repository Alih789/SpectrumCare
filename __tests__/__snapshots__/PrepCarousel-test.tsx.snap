// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrepCarousel renders PrepCarousel 1`] = `
<View
  style={
    {
      "height": "100%",
      "width": 750,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "margin": 10,
      }
    }
  >
    <View
      activeOffsetX={
        [
          -10,
          10,
        ]
      }
      enabled={true}
      onGestureEvent={[Function]}
    >
      <View
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "height": 1067,
              "width": 730,
            },
            {},
            {
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "height": 1067,
                "position": "absolute",
                "width": 730,
              },
              {
                "transform": [
                  {
                    "translateX": undefined,
                  },
                ],
              },
            ]
          }
          testID="__CAROUSEL_ITEM_0_NOT_READY__"
        />
        <View
          style={
            [
              {
                "height": 1067,
                "position": "absolute",
                "width": 730,
              },
              {
                "transform": [
                  {
                    "translateX": undefined,
                  },
                ],
              },
            ]
          }
          testID="__CAROUSEL_ITEM_1_NOT_READY__"
        />
        <View
          style={
            [
              {
                "height": 1067,
                "position": "absolute",
                "width": 730,
              },
              {
                "transform": [
                  {
                    "translateX": undefined,
                  },
                ],
              },
            ]
          }
          testID="__CAROUSEL_ITEM_2_NOT_READY__"
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "padding": 20,
        "width": 750,
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-start",
          "bottom": 20,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "position": "absolute",
          "width": 750,
        }
      }
    >
      <View
        style={
          [
            {
              "borderColor": "#ffffff",
              "borderRadius": 5.5,
              "borderWidth": 1,
              "height": 12,
              "marginBottom": 8,
              "marginHorizontal": 4,
              "marginTop": 10,
              "width": 12,
            },
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "borderColor": "#ffffff",
              "borderRadius": 5.5,
              "borderWidth": 1,
              "height": 12,
              "marginBottom": 8,
              "marginHorizontal": 4,
              "marginTop": 10,
              "width": 12,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          [
            {
              "borderColor": "#ffffff",
              "borderRadius": 5.5,
              "borderWidth": 1,
              "height": 12,
              "marginBottom": 8,
              "marginHorizontal": 4,
              "marginTop": 10,
              "width": 12,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
